### PR TITLE
use megahirt/wait instead of locally built wait

### PR DESCRIPTION
**Pull request by megahirt**
I built the wait image and pushed it to docker hub under my personal repository.  The wait image just includes docker-compose-wait, is simple and hasn't changed since @longrunningprocess created it, so I decided we could just build once and then reference it in our code as a simplification step.

This should hopefully simplify our building of the wait image, as well as resolve the conversation about where to build wait, as discussed in #891

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)

### CODE REVIEW

**Review by longrunningprocess — CHANGES_REQUESTED** [review]
let's use `sillsdev/web-languageforge:wait-latest` instead and blow away `docker/wait` altogether.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-:-MWjdcN17BGNc5-AvwZZ:btjp9t9)* status: 0 of 4 files reviewed, 6 unresolved discussions (waiting on @megahirt)

___
*[docker/Makefile, line 13 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdznB3Shz60h0nIxg:-MWjdznB3Shz60h0nIxh:b1fcw3u) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/Makefile#L13)):*
> ```Makefile
> 
> .PHONY: build-test-e2e
> build-test-e2e: wait
> ```

here's another one
___
*[docker/Makefile, line 21 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWje1DF5asR4Mwlk2z2:-MWje1DF5asR4Mwlk2z3:b1fcw3u) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/Makefile#L21)):*
> ```Makefile
> 
> .PHONY: build-test-php
> build-test-php: wait
> ```

here's another one
___
*[docker/Makefile, line 29 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdKF501dsk7tK7WvS:-MWjdKF501dsk7tK7WvT:b-qt0en0) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/Makefile#L29)):*
> ```Makefile
> 
> .PHONY: db-reset
> db-reset: wait
> ```

here's another one.
___
*[docker/app/Dockerfile, line 37 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdOev6mzQ8LkjEsVR:-MWjdOev6mzQ8LkjEsVS:b-lx5ppg) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/app/Dockerfile#L37)):*
> ```Dockerfile
> 
> # make wait available for container ochestration
> COPY --from=megahirt/wait /wait /wait
> ```

use `sillsdev/web-languageforge:wait-latest` instead
___
*[docker/test-e2e/Dockerfile, line 4 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdUZw15Cu79BfEaew:-MWjdUZw15Cu79BfEaex:b-lx5ppg) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/test-e2e/Dockerfile#L4)):*
> ```Dockerfile
> 
> # make wait available for container ochestration
> COPY --from=megahirt/wait /wait /wait
> ```

use `sillsdev/web-languageforge:wait-latest` instead
___
*[docker/wait/README.md, line 1 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjd__tEZWAtS_xWMdU:-MWjd__tEZWAtS_xWMdV:b-xtcugn) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/wait/README.md#L1)):*
> ```Markdown
> The wait image defined here is published to Docker Hub under the name "megahirt/wait".  We reference this image a few times in other Dockerfiles in this project.
> ```

let's remove this altogether

**Review by longrunningprocess — COMMENTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)* status: all files reviewed, 6 unresolved discussions (waiting on @megahirt)

**Review by megahirt — COMMENTED** [review]
Very helpful review - thank you.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-:-MX-m7X-AP8KDQ3qpEkw:bg4tk1j)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @longrunningprocess)

___
*[docker/Makefile, line 13 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdznB3Shz60h0nIxg:-MX-lzjcDFHy6-cfaUPZ:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/Makefile#L13)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

here's another one
</blockquote></details>

Done.
___
*[docker/Makefile, line 21 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWje1DF5asR4Mwlk2z2:-MX-m-HF-UDve72xPHfO:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/Makefile#L21)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

here's another one
</blockquote></details>

Done.
___
*[docker/Makefile, line 29 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdKF501dsk7tK7WvS:-MX-lz0U89ks_6acP7G4:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/Makefile#L29)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

here's another one.
</blockquote></details>

Done.
___
*[docker/app/Dockerfile, line 37 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdOev6mzQ8LkjEsVR:-MX-m1nM8GlLv4BaVLQ1:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/app/Dockerfile#L37)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

use `sillsdev/web-languageforge:wait-latest` instead
</blockquote></details>

Done.
___
*[docker/test-e2e/Dockerfile, line 4 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjdUZw15Cu79BfEaew:-MX-m2_z5sDhbqwkwvGT:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/test-e2e/Dockerfile#L4)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

use `sillsdev/web-languageforge:wait-latest` instead
</blockquote></details>

Done.
___
*[docker/wait/README.md, line 1 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjd__tEZWAtS_xWMdU:-MX-m5Jr3pSQqzIMcShU:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/wait/README.md#L1)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

let's remove this altogether
</blockquote></details>

Done.

**Review by megahirt — COMMENTED** [review]
> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)* status: all files reviewed, 6 unresolved discussions (waiting on @longrunningprocess)

**Review by longrunningprocess — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @megahirt)

___
*[docker/wait/README.md, line 1 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MWjd__tEZWAtS_xWMdU:-MX2E2jREUXaUFM6E1hg:balw8hg) ([raw file](https://github.com/sillsdev/web-languageforge/blob/8a5c573504c5d82d15b76e9f1b5e93496bc72c88/docker/wait/README.md#L1)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I meant to remove the entire `docker/wait` folder, not needed anymore.  I'll push up a quickie on that.

**Review by longrunningprocess — COMMENTED** [review]
I added some things that should help with the intentionality but it will require changing the TC build configs just before merge so we can ensure the builds don't break for all other build not based off this branch.  Also, please don't hesitate to decide we need to pair through this so we're not just creating work for each other every time.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-:-MX36W886TRT-1DPCfyM:b-l4znv0)* status: 2 of 5 files reviewed, all discussions resolved (waiting on @megahirt)

**Review by megahirt — COMMENTED** [review]
Thanks for this @longrunningprocess .  I think your changes make things more readable in the Makefile and in plain English so lets go with that.  The reason that I had been holding out on a separate build target for both e2e and php tests is my preference for CI is to have separate `build` and `test` steps.  When these are separate it's easier to see where a CI failure is: either the build step or tests failing.  It also makes the estimated time to completion more consistent when the build is separated from the testing step.  Test execution time should be predictable, while docker build time with image/layer caching can vary significantly. 
 However, maintaining this distinction requires repeated code in the Makefile.  And as I think about it I prefer the simpler of the two, so thank you for persisting on this :) :lgtm: 

I will go in an change the build config to a single "run tests" step and then we'll see this branch through.

> Reviewed 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-:-MXAUbLu8uC4mj3nqkZX:bmhulp2)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @megahirt)

___
*[README.md, line 95 at r3](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-MXAU5BS30QZcHuxooiW:-MXAU5BTEvGYCpIAXFpk:bhi1wja) ([raw file](https://github.com/sillsdev/web-languageforge/blob/fb52fa742e9eb95ac4f09b58784afa99f1d73b50/README.md#L95)):*
> ```Markdown
> ### Running E2E Tests
> 
> 1. `make e2e-tests`
> ```

this reads/sounds better.  Thank you.

**Review by megahirt — COMMENTED** [review]
I did a force-push to squash same-author commits before I merge.  Because there are two authors on this branch, I will do a traditional merge-commit to preserve authorship (no squash commit).

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893#-:-MXAWPut1z9ZsXU9MiHp:bkrzkfx)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @megahirt)

**Review by longrunningprocess — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 2 of 3 files at r3, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)* status: 4 of 5 files reviewed, all discussions resolved (waiting on @megahirt)

**Review by megahirt — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @megahirt)

**Review by longrunningprocess — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 2 of 3 files at r3, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/893)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @megahirt)
